src/cljs/rems/administration/create_form.cljs

```diff
@@ -124,26 +83,69 @@
                         :label (build-localized-string label languages)})})))
 
 (defn build-request [form languages]
-  (let [request {:organization (:organization form)
-                 :title (:title form)
-                 :fields (mapv #(build-request-field % languages) (:fields form))}]
-    (when (valid-request? request languages)
-      request)))
+  {:organization (:organization form)
+   :title (:title form)
+   :fields (mapv #(build-request-field % languages) (:fields form))})
+
+;;;; form validation
+
+(defn- validate-text-field [m key]
+  (when (str/blank? (get m key))
+    {key :t.form.validation/required}))
+
+(defn- validate-localized-text-field [m key languages]
+  {key (apply merge (mapv #(validate-text-field (get m key) %) languages))})
+
+(defn- validate-optional-localized-field [m key languages]
+  (let [validated (mapv #(validate-text-field (get m key) %) languages)]
+    ;; partial translations are not allowed
+    (when (not-empty (remove identity validated))
+      {key (apply merge validated)})))
+
+(defn- validate-maxlength [maxlength]
+  (when-not (str/blank? maxlength)
+    (let [parsed (parse-int maxlength)]
+      (when (or (neg? parsed) (nil? parsed))
+        {:maxlength :t.form.validation/invalid-value}))))
+
+(defn- validate-option [option id languages]
+  {id (merge (validate-text-field option :key)
+             (validate-localized-text-field option :label languages))})
+
+(defn- validate-options [options languages]
+  {:options (apply merge (mapv #(validate-option %1 %2 languages) options (range)))})
+
+(defn- validate-field [field id languages]
+  {id (merge (validate-text-field field :type)
+             (validate-localized-text-field field :title languages)
+             (validate-optional-localized-field field :input-prompt languages)
+             (validate-maxlength (:maxlength field))
+             (validate-options (:options field) languages))})
+
+(defn- validate-form [form languages]
+  (merge (validate-text-field form :organization)
+         (validate-text-field form :title)
+         {:fields (apply merge (mapv #(validate-field %1 %2 languages) (form :fields) (range)))}))
+
+(defn- nil-vals? [x]
+  (cond (map? x) (every? nil-vals? (vals x)) (nil? x) true :else false))
```

Comment:
I think this `cond` should be split to multiple lines but perhaps this whole fn should be unnecessary?

Reply:
Done, the function is gone